specification/under_development/output_schemas/Labeled_And_Async.md

```diff
@@ -1,2 +1,241 @@
 # Labeled Format for Asynchronous Output Generation
 
+The labeled format for asynchronous output generation is the same as the [ordered format][] with the following changes:
+  - `OUTPUT` records `TUPLE`, `ARRAY`, `RESULT`, `INT`, `BOOL`, and `DOUBLE` types have a fourth field indicating the label of the record.
+
+Labels are needed for reconstruction of asynchronous/parallel computation output and are assigned by the front-end QIR generator. Order is not important for the `OUTPUT` entries within a `START`/`END` block; however, the responsibility of reconstructing the output based on the defined labeling scheme belongs to the party permforming the output labeling. The usage of `t0_0a` and `t2_2a` (and other values) are examples of a labeling scheme, and are only used as an example.
+
+Consumers of the QIR need to map the associated labeles for each recording call to its output entry label.
+
+Example log for a single shot:
+
+```log
+START
+METADATA\tmetadata1_name_only
+METADATA\tmetadata2_name\tmetadata2_value
+METADATA\tmetadata3_name\tmetadata3_value
+OUTPUT\tTUPLE\t2\t0_t
+OUTPUT\tARRAY\t4\t1_t0a
+OUTPUT\tRESULT\t0\t2_t0a0r
+OUTPUT\tRESULT\t0\t3_t0a1r
+OUTPUT\tRESULT\t0\t4_t0a2r
+OUTPUT\tRESULT\t0\t5_t0a3r
+OUTPUT\tTUPLE\t3\t6_t1t
+OUTPUT\tBOOL\ttrue\t7_t1t0b
+OUTPUT\tINT\t42\t8_t1t1i
+OUTPUT\tDOUBLE\t3.1415\t9_t1t2d
+END\t0
+```
+
+## Declared Runtime Functions 
+
+Each of these runtime functions are meant as signifiers to the provider as to how raw device results should be collected into the common output format. For simulation or future environments with full runtime support, these functions can be linked to implementations that directly perform the recording of output to the relevant stream. Each of these functions follow the naming pattern `__quantum__rt__*_record_output` where the initial part indicates the type of output to be recorded.  
+
+### Primitive Result Records 
+
+```llvm
+void @__quantum__rt__result_record_output(%Result*, i8*)
+```
+
+Adds a measurement result to the generated output. It produces output records of exactly `"OUTPUT\tRESULT\t0\tlabel"` or `"OUTPUT\tRESULT\t1\tlabel"`. The second parameter defines a string label for the result value which is included in the output (`label`).
```

Comment:
While I understand why the method signatures are specialized, I wouldn't mind a sentence explaining why RESULT and BOOL _outputs_ aren't emitted as INT.

Reply:
Addressed in the notes that were just pushed.